app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/startregistration/StartRegistrationScreen.kt

```diff
@@ -0,0 +1,426 @@
+                onOptionSelected = remember(viewModel) {
+                    { viewModel.trySendAction(EnvironmentTypeSelect(it)) }
+                },
+                modifier = Modifier
+                    .testTag("RegionSelectorDropdown")
+                    .padding(horizontal = 16.dp)
+                    .fillMaxWidth(),
+            )
+            Spacer(modifier = Modifier.height(16.dp))
+            BitwardenTextField(
+                label = stringResource(id = R.string.name),
+                value = state.nameInput,
+                onValueChange = remember(viewModel) {
+                    { viewModel.trySendAction(NameInputChange(it)) }
+                },
+                modifier = Modifier
+                    .testTag("NameEntry")
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp),
+            )
+            Spacer(modifier = Modifier.height(16.dp))
+
+            if (state.selectedEnvironmentType != Environment.Type.SELF_HOSTED) {
+                ReceiveMarketingEmailsSwitch(
+                    isChecked = state.isReceiveMarketingEmailsToggled,
+                    onCheckedChange = remember(viewModel) {
+                        {
+                            viewModel.trySendAction(
+                                ReceiveMarketingEmailsToggle(
+                                    it,
+                                ),
+                            )
+                        }
+                    },
+                    onUnsubscribeClick = remember(viewModel) {
+                        { viewModel.trySendAction(UnsubscribeMarketingEmailsClick) }
+                    },
+                )
+                Spacer(modifier = Modifier.height(16.dp))
+            }
+
+            BitwardenFilledButton(
+                label = stringResource(id = R.string.continue_text),
+                onClick = remember(viewModel) {
+                    { viewModel.trySendAction(ContinueClick) }
+                },
+                isEnabled = state.isContinueButtonEnabled,
+                modifier = Modifier
+                    .testTag("ContinueButton")
+                    .padding(horizontal = 16.dp)
+                    .fillMaxWidth(),
+            )
+            Spacer(modifier = Modifier.height(16.dp))
+            TermsAndPrivacyText(
+                onTermsClick = remember(viewModel) {
+                    { viewModel.trySendAction(TermsClick) }
+                },
+                onPrivacyPolicyClick = remember(viewModel) {
+                    { viewModel.trySendAction(PrivacyPolicyClick) }
+                },
+            )
+            Spacer(modifier = Modifier.navigationBarsPadding())
+        }
+    }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Suppress("LongMethod")
+@Composable
+private fun TermsAndPrivacyText(
+    onTermsClick: () -> Unit,
+    onPrivacyPolicyClick: () -> Unit,
+) {
+    val annotatedLinkString: AnnotatedString = buildAnnotatedString {
```

Comment:
💭 Would it make sense to move the [CreateAnnotatedString](https://github.com/bitwarden/android/pull/3621/files#diff-5130896a958fba54a263f85d0ef3a7b053b0d6e2ad8f607de21ef9dbc1268842R202) fwd into this change? And use that.

Reply:
also as a side note for `@Composable` functions which return a value the convention is to use `camelCase` and `PascalCase` for they return `Unit`